core/lib/snapshots_applier/src/lib.rs

```diff
@@ -355,6 +371,23 @@ impl<'a> SnapshotsApplier<'a> {
                 .map_err(|err| {
                     SnapshotsApplierError::db(err, "failed persisting initial recovery status")
                 })?;
+            storage_transaction
```

Comment:
it simplifies dbPruner logic, it doesn't have to rely on snapshots creator, same goes for API server, it can only rely on pruning_info instead of relying on both snapshots_recovery and pruning_info

Reply:
I strongly believe such change would do more harm than good, and that treating snapshots recovery as a form of pruning is a good abstraction